### PR TITLE
[v0.10.7]: Fix the gentle reset logic

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Sets block stall deadline after a successful sync check.

The reason for that change is that when we perform a gentle reset, the check for the stall deadlines happens before a begin block is received and we end up in a loop, e.g.:

```
.
.
.
[2023-06-21 09:13:37,571] [INFO] [agent] Performing a gentle reset.
[2023-06-21 09:13:37,657] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:37,658] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:37,658] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:37,658] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:37,958] [INFO] [agent] Checking sync...
[2023-06-21 09:13:37,959] [INFO] [agent] Checking status @ http://localhost:26657/status
[2023-06-21 09:13:37,965] [ERROR] [agent] Tendermint not accepting transactions yet, trying again!
[2023-06-21 09:13:40,965] [INFO] [agent] Checking status @ http://localhost:26657/status
[2023-06-21 09:13:40,972] [INFO] [agent] local height == remote == 317; Sync complete...
[2023-06-21 09:13:40,972] [INFO] [agent] Applying tendermint fix finished.
[2023-06-21 09:13:40,973] [WARNING] [agent] The local deadline for the next `begin_block` request from the Tendermint node has expired! Trying to reset local Tendermint node as there could be something wrong with the communication.
[2023-06-21 09:13:40,973] [INFO] [agent] Performing a gentle reset.
[2023-06-21 09:13:41,876] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:41,877] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:41,877] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:41,877] [INFO] [agent] connection at EOF, stop receiving loop.
[2023-06-21 09:13:42,059] [INFO] [agent] Checking sync...
[2023-06-21 09:13:42,059] [INFO] [agent] Checking status @ http://localhost:26657/status
[2023-06-21 09:13:42,064] [ERROR] [agent] Tendermint not accepting transactions yet, trying again!
[2023-06-21 09:13:45,065] [INFO] [agent] Checking status @ http://localhost:26657/status
[2023-06-21 09:13:45,070] [INFO] [agent] local height == remote == 317; Sync complete...
[2023-06-21 09:13:45,071] [INFO] [agent] Applying tendermint fix finished.
[2023-06-21 09:13:45,072] [WARNING] [agent] The local deadline for the next `begin_block` request from the Tendermint node has expired! Trying to reset local Tendermint node as there could be something wrong with the communication.
.
.
.
```

If we immediately set the stall deadline after a successful sync, then the check will be done using the new deadline.

This PR also performs some unrelated cleanup on 4b3955e632a2584154ff04bcd7f70baf96ecc02a.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [x] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

## Further comments

n/a

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/behaviour_utils.py [hunk 1]
```diff
@@ -760,6 +760,9 @@ def _check_sync(
                         f"local height == remote == {local_height}; Sync complete..."
                     )
                     self.round_sequence.end_sync()
+                    # we set the block stall deadline here because we pinged the /status endpoint
+                    # and received a response from tm, which means that the communication is fine
+                    self.round_sequence.set_block_stall_deadline()
```
Comment (Adamantios) [inline packages/valory/skills/abstract_round_abci/behaviour_utils.py:765]: This is the fix. The rest is mostly unrelated cleanup, done in 4b3955e632a2584154ff04bcd7f70baf96ecc02a.

**Review by 0xArdi — APPROVED** [review]

**Review by dvilelaf — APPROVED** [review]
It seems main needs to be merged, but LGTM.
